maze/README.rst

```diff
@@ -0,0 +1,79 @@
+Demo of Maze Solving
+====================
+Getting the D-Wave quantum computer to solve a maze!
+
+The following code takes a simple and familiar problem---solving a maze---and demonstrates the steps
+of submitting such problems to the quantum computer.
+
+Code Overview
+-------------
+The solution technique is to construct a set of constraints that enforces the rules of moving
+through a maze. These constraints are then converted by Ocean software tools to a binary
+quadratic model (BQM) that can then be solved with a D-Wave quantum computer. The solution that gets
+returned by the quantum computer is the path needed to get through the maze.
+
+There are several constraints involved with a maze:
+
+ - Valid path moves (i.e., if the path enters a grid point, it must also leave said grid point)
+ - Path has a specific start and end position
+ - Path cannot pass maze borders
+ - Path cannot pass internal walls
+ 
+Each of these constraints are implemented by the :code:`maze` functions:
+:code:`_apply_valid_move_constraint()`, :code:`_set_start_and_end()`, :code:`_set_borders()`, and
+:code:`_set_inner_walls()`, respectively. These functions are called when the user calls
+:code:`get_maze_bqm(..)` in the Example section below.
+
+Code Specifics
+--------------
+The maze is a rectangular grid. The path segments (aka edges) that can be formed in this grid are
+described with respect to a grid point. For example, the edge labelled :code:`'1,0w'`:
+
+ - :code:`1,0` refers to grid point on row 1, column 0
+ - :code:`w` refers to "west"
+
+Hence, if you imagine a compass that is centered at position :code:`1,0`, the edge :code:`'1,0w'`
+is the segment along the west direction of this compass.
```

Comment:
@JoelPasvolsky I've removed the "spoke" part of the analogy. Does it make more sense now?